skipper_test.go

```diff
@@ -109,6 +109,28 @@ func TestOptionsFilterRegistry(t *testing.T) {
 	assert.NotContains(t, fr, filters.BearerInjectorName)
 }
 
+func TestOptionsTracerInstanceOverridesOpenTracing(t *testing.T) {
+	tracer := &tracingtest.Tracer{}
+	o := Options{
+		Tracer:      tracer,
+		OpenTracing: []string{"noop"},
+	}
+
+	tr, err := o.tracerInstance()
+	assert.Nil(t, err)
+	assert.Same(t, tr, tracer)
+}
+
+func TestOptionsTracerInstanceFallbacksToOpenTracingWhenTracerIsNil(t *testing.T) {
```

Comment:
Maybe just name it `TestOptionsOpenTracing`.

Lets also add a test (`TestOptionsDefaultTracer`) for the fallback tracer from empty `Options{}`

Reply:
Looks like we overlooked a case where tracer can still be nil. This is when `o.Tracer = nil`, `o.OpenTracing = nil` & `pluginDirs = nil`.

Reply:
Hm, ok, then I guess we can return (nil, nil), have a test for it and restore nil check for tee filter override

Reply:
alternatively we can also send back a noop tracer. WDYT?

Reply:
I think we should change `LoadTracingPlugin` to return error if no plugin was loaded.

Reply:
I propose https://github.com/zalando/skipper/pull/2525#discussion_r1299939623